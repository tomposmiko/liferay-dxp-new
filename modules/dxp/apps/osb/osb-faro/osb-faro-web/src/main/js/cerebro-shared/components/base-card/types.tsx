export type Router = {
	params: object;
	query: object;
};

export type Context = {
	filters: object;
	router: Router;
};
