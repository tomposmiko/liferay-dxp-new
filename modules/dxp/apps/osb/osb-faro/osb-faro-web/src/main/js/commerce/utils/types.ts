export type Trend = {
	trendClassification: 'POSITIVE' | 'NEGATIVE' | 'NEUTRAL';
	percentage: number;
};
