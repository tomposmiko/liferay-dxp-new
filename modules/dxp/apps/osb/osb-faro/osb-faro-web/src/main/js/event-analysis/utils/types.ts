import {OrderByDirections} from 'shared/util/constants';

export enum AttributeOwnerTypes {
	Account = 'ACCOUNT',
	Event = 'EVENT',
	Individual = 'INDIVIDUAL',
	Session = 'SESSION'
}

export enum AttributeTypes {
	All = 'ALL',
	Global = 'GLOBAL',
	Local = 'LOCAL'
}

export enum CalculationTypes {
	Average = 'AVERAGE',
	Total = 'TOTAL',
	Unique = 'UNIQUE'
}

export enum DataTypes {
	Boolean = 'BOOLEAN',
	Date = 'DATE',
	Duration = 'DURATION',
	Number = 'NUMBER',
	String = 'STRING'
}

export enum DateGroupings {
	Day = 'DAY',
	Month = 'MONTH',
	Year = 'YEAR'
}

export enum EventTypes {
	All = 'ALL',
	Custom = 'CUSTOM',
	Default = 'DEFAULT'
}

export enum Operators {
	Between = 'between',
	Contains = 'contains',
	NotContains = 'notContains',
	EQ = 'eq',
	GT = 'gt',
	LT = 'lt',
	NE = 'ne'
}

export type Attribute = {
	dataType: DataTypes;
	description?: string;
	displayName?: string;
	id: string;
	name: string;
	recentValues?: {
		lastSeenDate: string;
		value: string;
	}[];
	sampleValue?: string;
	type: AttributeTypes;
};

export type Attributes = {[key: string]: Attribute};

export type BlockedCustomEvent = {
	hidden: boolean;
	id: string;
	name: string;
	lastSeenDate: string;
	lastSeenURL: string;
};

export type Breakdown = {
	attributeId: string;
	attributeType: AttributeOwnerTypes;
	binSize: number;
	dataType: DataTypes;
	dateGrouping: DateGroupings;
	description?: string;
	displayName: string;
	id?: string;
	sortType: OrderByDirections;
};

export type Breakdowns = {[key: string]: Breakdown};

export type Event = {
	description?: string;
	displayName?: string;
	eventAttributeDefinitions?: Attribute[];
	hidden: boolean;
	id: string;
	name: string;
	type: EventTypes;
};

export type Filter = {
	attributeId: string;
	attributeType: AttributeOwnerTypes;
	dataType: DataTypes;
	description?: string;
	displayName: string;
	id?: string;
	operator: Operators;
	values: string[];
};

export type Filters = {[key: string]: Filter};

export interface IBreakdownProps {
	attributeId: string;
	attributeOwnerType: AttributeOwnerTypes;
	breakdown?: Breakdown;
	description?: string;
	displayName: string;
	onSubmit: (breakdown: Breakdown) => void;
}

export interface IFilterProps {
	attributeId: string;
	attributeOwnerType: AttributeOwnerTypes;
	description?: string;
	displayName: string;
	eventId?: string;
	filter?: Filter;
	onSubmit: (filter: Filter) => void;
}

export type BreakdownDataItem = {
	breakdownItems?: BreakdownDataItem[];
	leafNode: boolean;
	name: string;
	previousValue?: number;
	value: number;
};

export type BreakdownData = {
	breakdownItems: BreakdownDataItem[];
	count: number;
	page: number;
	previousValue?: number;
	value: number;
};

export type ParsedBreakdownItem = {
	events: BreakdownDataItem[];
	index: string;
	breakdown0: BreakdownDataItem & {rowSpan: number};
	[key: string]:
		| (BreakdownDataItem & {rowSpan: number})
		| BreakdownDataItem[]
		| string;
};

export type ParsedBreakdownData = ParsedBreakdownItem[];
