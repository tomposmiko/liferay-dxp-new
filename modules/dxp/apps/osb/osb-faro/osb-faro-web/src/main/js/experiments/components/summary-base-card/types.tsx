import React from 'react';

export type Modal = {
	Component: any;
	props: Object;
	title?: string;
};

export type Alert = {
	description?: string;
	symbol?: string;
	title: string;
};

export type ButtonProps = {
	['data-tooltip']?: boolean;
	disabled?: boolean;
	label: string;
	symbol?: string;
	title?: string;
};

export type Description = {
	className: string;
};

export type Header = {
	modals?: Array<Modal>;
	cardModals?: Array<Modal>;
	Description?: React.FC;
	title: string;
};

export type Setup = {
	current: number;
	steps: Array<Step>;
	subtitle?: string;
	title: string;
};

export type Summary = {
	description?: string;
	subtitle?: string;
	title: string;
};

export type Status =
	| 'completed'
	| 'draft'
	| 'finished'
	| 'running'
	| 'scheduled'
	| 'terminated';

export type Step = {
	modal?: Modal;
	buttonProps: ButtonProps;
	Description: React.FC<Description>;
	experimentLink?: string;
	link?: string;
	title: string;
};
