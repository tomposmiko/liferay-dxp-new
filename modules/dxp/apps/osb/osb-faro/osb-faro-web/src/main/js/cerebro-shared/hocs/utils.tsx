import NoResultsDisplay from 'shared/components/NoResultsDisplay';
import React from 'react';
import {Sizes} from 'shared/util/constants';

interface IIconProps {
	border?: boolean;
	size?: Sizes;
	symbol: string;
}

type TWithError = () => (
	Component: React.JSXElementConstructor<any>
) => (props: any) => React.ReactElement;

const withError: TWithError = () => Component => ({
	error,
	errorMessage,
	...props
}) => {
	if (error) {
		return (
			<NoResultsDisplay
				title={
					errorMessage ||
					Liferay.Language.get('sorry-an-error-occurred')
				}
			/>
		);
	}

	return <Component {...props} />;
};

type TWithEmpty = (params?: {
	emptyDescription?: string | React.ReactElement;
	emptyIcon?: IIconProps;
	emptyTitle?: string;
	primary?: boolean;
}) => (
	Component: React.JSXElementConstructor<any>
) => (props: any) => React.ReactElement;

const withEmpty: TWithEmpty = ({
	emptyDescription,
	emptyIcon,
	emptyTitle,
	primary
} = {}) => Component => ({
	items,
	noResultsRenderer,
	query,
	total,
	...otherProps
}) => {
	if (items && !items.length && !total) {
		if (query) {
			return (
				<NoResultsDisplay
					description={Liferay.Language.get(
						'please-try-a-different-search-term'
					)}
					icon={{
						border: false,
						size: Sizes.XXXLarge,
						symbol: 'ac-no-results-found'
					}}
					title={Liferay.Language.get('there-are-no-results-found')}
				/>
			);
		} else if (noResultsRenderer) {
			return noResultsRenderer;
		}

		return (
			<NoResultsDisplay
				description={emptyDescription}
				icon={emptyIcon}
				primary={primary}
				title={emptyTitle}
			/>
		);
	}

	return (
		<Component {...otherProps} items={items} query={query} total={total} />
	);
};

export {withEmpty, withError};
