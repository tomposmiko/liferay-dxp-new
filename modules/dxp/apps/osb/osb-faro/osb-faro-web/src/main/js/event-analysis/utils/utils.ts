import moment from 'moment';
import {
	Attribute,
	AttributeOwnerTypes,
	Breakdown,
	BreakdownData,
	BreakdownDataItem,
	DataTypes,
	DateGroupings,
	Filter,
	Operators,
	ParsedBreakdownData,
	ParsedBreakdownItem
} from './types';
import {formatTime} from 'shared/util/time';
import {formatUTCDate} from 'shared/util/date';
import {OrderByDirections} from 'shared/util/constants';

const DEFAULT_DATE_GROUPING = DateGroupings.Month;
const DEFAULT_DURATION_BIN = 60000;
const DEFAULT_NUMBER_BIN = 10;

const ATTRIBUTE_TYPE_LABEL_MAP = {
	[AttributeOwnerTypes.Account]: Liferay.Language.get('account'),
	[AttributeOwnerTypes.Event]: Liferay.Language.get('event'),
	[AttributeOwnerTypes.Individual]: Liferay.Language.get('individual'),
	[AttributeOwnerTypes.Session]: Liferay.Language.get('session')
};

export const BOOLEAN_OPTIONS = ['true', 'false'];

export const DATE_GROUPING_OPTIONS = [
	DateGroupings.Day,
	DateGroupings.Month,
	DateGroupings.Year
];

export const DATE_OPTIONS = [
	Operators.EQ,
	Operators.LT,
	Operators.GT,
	Operators.Between
];

export const DURATION_OPTIONS = [Operators.GT, Operators.LT];

export const NUMBER_OPTIONS = [Operators.GT, Operators.LT, Operators.Between];

export const STRING_OPTIONS = [
	Operators.Contains,
	Operators.NotContains,
	Operators.EQ,
	Operators.NE
];

export const BOOLEAN_LABELS_MAP = {
	false: Liferay.Language.get('false'),
	true: Liferay.Language.get('true')
};

export const DATA_TYPE_ICONS_MAP = {
	[DataTypes.Boolean]: 'check',
	[DataTypes.Date]: 'date',
	[DataTypes.Duration]: 'time',
	[DataTypes.Number]: 'integer',
	[DataTypes.String]: 'text'
};

export const DATA_TYPE_LABELS_MAP = {
	[DataTypes.Boolean]: Liferay.Language.get('boolean'),
	[DataTypes.Date]: Liferay.Language.get('date'),
	[DataTypes.Duration]: Liferay.Language.get('duration'),
	[DataTypes.Number]: Liferay.Language.get('number'),
	[DataTypes.String]: Liferay.Language.get('string')
};

export const DATE_GROUPING_LABELS_MAP = {
	[DateGroupings.Day]: Liferay.Language.get('date'),
	[DateGroupings.Month]: Liferay.Language.get('month'),
	[DateGroupings.Year]: Liferay.Language.get('year')
};

export const DATE_OPERATOR_LABELS_MAP = {
	[Operators.Between]: '-',
	[Operators.EQ]: '=',
	[Operators.GT]: Liferay.Language.get('after-fragment'),
	[Operators.LT]: Liferay.Language.get('before-fragment')
};

export const DATE_OPERATOR_LONGHAND_LABELS_MAP = {
	[Operators.Between]: Liferay.Language.get('is-between-fragment'),
	[Operators.EQ]: Liferay.Language.get('is-fragment'),
	[Operators.GT]: Liferay.Language.get('after-fragment'),
	[Operators.LT]: Liferay.Language.get('before-fragment')
};

export const DURATION_OPERATOR_LABELS_MAP = {
	[Operators.GT]: '>',
	[Operators.LT]: '<'
};

export const DURATION_OPERATOR_LONGHAND_LABELS_MAP = {
	[Operators.GT]: Liferay.Language.get('is-greater-than-fragment'),
	[Operators.LT]: Liferay.Language.get('is-less-than-fragment')
};

export const NUMBER_OPERATOR_LABELS_MAP = {
	[Operators.Between]: '-',
	[Operators.GT]: '>',
	[Operators.LT]: '<'
};

export const NUMBER_OPERATOR_LONGHAND_LABELS_MAP = {
	[Operators.Between]: Liferay.Language.get('between-fragment'),
	[Operators.GT]: Liferay.Language.get('is-greater-than-fragment'),
	[Operators.LT]: Liferay.Language.get('is-less-than-fragment')
};

export const STRING_OPERATOR_LABELS_MAP = {
	[Operators.Contains]: Liferay.Language.get('contains-fragment'),
	[Operators.NotContains]: Liferay.Language.get('not-contains-fragment'),
	[Operators.EQ]: Liferay.Language.get('is-fragment'),
	[Operators.NE]: Liferay.Language.get('is-not-fragment')
};

const getBooleanDisplay = (
	attribute: Attribute,
	{attributeType, values: [value]}: Filter
): [string, string] => [
	getBreakdownDisplay(attribute, attributeType).join(' | '),
	BOOLEAN_LABELS_MAP[String(value)]
];

const getDateDisplay = (
	attribute: Attribute,
	{attributeType, operator, values: [startDate, endDate]}: Filter
): [string, string] => {
	const operatorLabel = DATE_OPERATOR_LABELS_MAP[operator];

	const formattedStartDate = formatUTCDate(startDate as string, 'll');

	const breakdownValue =
		operator === Operators.Between
			? `${formattedStartDate} ${operatorLabel} ${formatUTCDate(
					endDate as string,
					'll'
			  )}`
			: `${operatorLabel} ${formattedStartDate}`;

	return [
		getBreakdownDisplay(attribute, attributeType).join(' | '),
		`${breakdownValue}`
	];
};

export const getBreakdownDisplay = (
	{displayName, name}: Attribute,
	attributeType: AttributeOwnerTypes
): [string, string] => [
	ATTRIBUTE_TYPE_LABEL_MAP[attributeType],
	displayName || name
];

const getDurationDisplay = (
	attribute: Attribute,
	{attributeType, operator, values: [value]}: Filter
): [string, string] => {
	const duration = formatTime(Number(value));

	return [
		getBreakdownDisplay(attribute, attributeType).join(' | '),
		`${DURATION_OPERATOR_LABELS_MAP[operator]} ${duration}`
	];
};

const getNumberDisplay = (
	attribute: Attribute,
	{attributeType, operator, values: [start, end]}: Filter
): [string, string] => {
	const operatorLabel = NUMBER_OPERATOR_LABELS_MAP[operator];

	const breakdownValue =
		operator === Operators.Between
			? `${start} ${operatorLabel} ${end}`
			: `${operatorLabel} ${start}`;

	return [
		getBreakdownDisplay(attribute, attributeType).join(' | '),
		breakdownValue
	];
};

const getStringDisplay = (
	attribute: Attribute,
	{attributeType, operator, values}: Filter
): [string, string] => [
	getBreakdownDisplay(attribute, attributeType).join(' | '),
	`${STRING_OPERATOR_LABELS_MAP[operator]} "${values}"`
];

const FILTER_DISPLAY_MAP = {
	[DataTypes.Boolean]: getBooleanDisplay,
	[DataTypes.Date]: getDateDisplay,
	[DataTypes.Duration]: getDurationDisplay,
	[DataTypes.Number]: getNumberDisplay,
	[DataTypes.String]: getStringDisplay
};

export const getFilterDisplay = (
	attribute: Attribute,
	filter: Filter
): [string, string] => {
	const displayFn = FILTER_DISPLAY_MAP[filter.dataType];

	return displayFn(attribute, filter);
};

export const isAttribute = (item: Attribute | Event): boolean =>
	(item as Attribute).dataType !== undefined;

export const createBooleanBreakdown = ({
	attributeId,
	attributeType,
	description,
	displayName
}): Breakdown => ({
	attributeId,
	attributeType,
	binSize: null,
	dataType: DataTypes.Boolean,
	dateGrouping: null,
	description,
	displayName,
	sortType: OrderByDirections.Descending
});

export const createDateBreakdown = ({
	attributeId,
	attributeType,
	dateGrouping = DEFAULT_DATE_GROUPING,
	description,
	displayName
}): Breakdown => ({
	attributeId,
	attributeType,
	binSize: null,
	dataType: DataTypes.Date,
	dateGrouping,
	description,
	displayName,
	sortType: OrderByDirections.Descending
});

export const createDurationBreakdown = ({
	attributeId,
	attributeType,
	binSize = DEFAULT_DURATION_BIN,
	description,
	displayName
}): Breakdown => ({
	attributeId,
	attributeType,
	binSize,
	dataType: DataTypes.Duration,
	dateGrouping: null,
	description,
	displayName,
	sortType: OrderByDirections.Descending
});

export const createNumberBreakdown = ({
	attributeId,
	attributeType,
	binSize = DEFAULT_NUMBER_BIN,
	description,
	displayName
}): Breakdown => ({
	attributeId,
	attributeType,
	binSize,
	dataType: DataTypes.Number,
	dateGrouping: null,
	description,
	displayName,
	sortType: OrderByDirections.Descending
});

export const createStringBreakdown = ({
	attributeId,
	attributeType,
	description,
	displayName
}): Breakdown => ({
	attributeId,
	attributeType,
	binSize: null,
	dataType: DataTypes.String,
	dateGrouping: null,
	description,
	displayName,
	sortType: OrderByDirections.Descending
});

export const BREAKDOWN_FNS_MAP = {
	[DataTypes.Boolean]: createBooleanBreakdown,
	[DataTypes.Date]: createDateBreakdown,
	[DataTypes.Duration]: createDurationBreakdown,
	[DataTypes.Number]: createNumberBreakdown,
	[DataTypes.String]: createStringBreakdown
};

export const getRowSpan = (breakdownItems: BreakdownDataItem[]): number => {
	let rowSpan = breakdownItems.length || 1;

	breakdownItems.forEach(({breakdownItems, leafNode}) => {
		if (!leafNode) {
			rowSpan = rowSpan + (getRowSpan(breakdownItems) - 1);
		}
	});

	return rowSpan;
};

export const formatDateName = (
	name: string,
	dateGrouping: DateGroupings
): string => {
	switch (dateGrouping) {
		case DateGroupings.Day:
			return moment(name, 'YYYY-MM-DD').format('ll');
		case DateGroupings.Month:
			return moment(name, 'YYYY-MM').format('MMM YYYY');
		case DateGroupings.Year:
		default:
			return name;
	}
};

export const formatDurationName = (name: string): string => {
	const [durationStart, durationEnd] = name.split('-');

	return `${formatTime(Number(durationStart))} - ${formatTime(
		Number(durationEnd)
	)}`;
};

export const formatBreakdownNameByDataType = (
	name: string,
	breakdown: Breakdown
): string | number => {
	if (name === 'undefined') {
		return name;
	}

	switch (breakdown?.dataType) {
		case DataTypes.Date:
			return formatDateName(name, breakdown.dateGrouping);
		case DataTypes.Duration:
			return formatDurationName(name);
		case DataTypes.Boolean:
		case DataTypes.String:
		case DataTypes.Number:
		default:
			return name;
	}
};

export const parseBreakdownData = (
	{breakdownItems}: BreakdownData | BreakdownDataItem,
	orderedBreakdowns: Breakdown[],
	rows: ParsedBreakdownData = [{index: '0'} as ParsedBreakdownItem],
	level: number = 0
): ParsedBreakdownData => {
	breakdownItems.forEach(data => {
		const {
			breakdownItems: nextBreakdownItems,
			leafNode: isLeafCurrentNode,
			name,
			...node
		} = data;

		const currentRowIndex = rows.length - 1;

		if (isLeafCurrentNode && level === 0) {
			Object.assign(rows[currentRowIndex], {
				events: [data]
			});

			return;
		}

		const isLeafNextNode =
			nextBreakdownItems.length > 0 && nextBreakdownItems[0].leafNode;

		Object.assign(rows[currentRowIndex], {
			[`breakdown${level}`]: {
				...node,
				name: isLeafCurrentNode
					? name
					: formatBreakdownNameByDataType(
							name,
							orderedBreakdowns[level]
					  ),
				rowSpan:
					!isLeafCurrentNode && !isLeafNextNode
						? getRowSpan(nextBreakdownItems)
						: 1
			},
			index: currentRowIndex
		});

		if (!nextBreakdownItems.length) {
			rows.push({} as ParsedBreakdownItem);
		}

		if (!isLeafNextNode) {
			parseBreakdownData(data, orderedBreakdowns, rows, level + 1);
		} else {
			Object.assign(rows[currentRowIndex], {
				events: nextBreakdownItems
			});

			rows.push({} as ParsedBreakdownItem);
		}
	});

	return level === 0 && !breakdownItems.length
		? rows
		: rows.filter(obj => Object.keys(obj).length !== 0);
};

export const getMaxEventValue = (parsedData, compareToPrevious: boolean) =>
	parsedData.reduce(
		(prev, {events = []}) =>
			events.reduce(
				(
					prev2,
					{
						breakdownItems: segments = [],
						previousValue = 0,
						value = 0
					}
				) =>
					segments.length <= 1
						? Math.max(
								value,
								prev2,
								compareToPrevious && previousValue
						  )
						: segments.reduce(
								(prev3, {previousValue, value}) =>
									Math.max(
										value,
										prev3,
										compareToPrevious && previousValue
									),
								prev2
						  ),
				prev
			),
		0
	);
