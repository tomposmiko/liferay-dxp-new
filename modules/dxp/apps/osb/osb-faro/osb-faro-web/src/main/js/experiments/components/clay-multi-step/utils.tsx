export const getStatus = (stepNumber: number, current: number) => {
	let status = 'wait';

	if (stepNumber === current) {
		status = 'active';
	} else if (stepNumber < current) {
		status = 'complete';
	}

	return status;
};
