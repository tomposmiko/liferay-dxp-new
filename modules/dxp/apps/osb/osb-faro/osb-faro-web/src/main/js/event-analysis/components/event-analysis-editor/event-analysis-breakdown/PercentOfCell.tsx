import React from 'react';
import {BreakdownDataItem} from 'event-analysis/utils/types';
import {get, round} from 'lodash';
import {getPercentage} from 'shared/util/util';

interface IPercentOfCellProps extends React.HTMLAttributes<HTMLElement> {
	compareToPrevious?: boolean;
	events: BreakdownDataItem[];
	totalValue: number;
}

const PercentOfCell: React.FC<IPercentOfCellProps> = ({
	compareToPrevious = false,
	events = [],
	totalValue
}) => {
	const isComparingEvent = events.length > 1;
	const isComparingSegment = get(events[0], 'breakdownItems', []).length > 1;

	const data = isComparingSegment
		? getItems(events[0].breakdownItems, compareToPrevious)
		: getItems(events, compareToPrevious);

	return (
		<>
			<ul className='percentage-column'>
				{data.map(({value}, i) => (
					<li key={i}>{`${round(
						getPercentage(value, totalValue),
						2
					)}%`}</li>
				))}
			</ul>

			{isComparingSegment && isComparingEvent && (
				<ul className='percentage-column'>
					{getItems(events[1].breakdownItems, compareToPrevious).map(
						({value}, i) => (
							<li key={i}>{`${round(
								getPercentage(value, totalValue),
								2
							)}%`}</li>
						)
					)}
				</ul>
			)}
		</>
	);
};

const getItems = (
	events: BreakdownDataItem[],
	compareToPrevious: boolean
): {
	value: number;
}[] => {
	const data = [];

	events.forEach(({previousValue, value}) => {
		data.push({
			value
		});

		if (compareToPrevious) {
			data.push({
				value: previousValue
			});
		}
	});

	return data;
};

export default PercentOfCell;
