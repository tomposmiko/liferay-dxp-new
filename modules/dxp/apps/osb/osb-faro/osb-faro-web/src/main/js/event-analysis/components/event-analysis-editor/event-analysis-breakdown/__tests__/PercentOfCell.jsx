import PercentOfCell from '../PercentOfCell';
import React from 'react';
import {DataTypes} from 'event-analysis/utils/types';
import {mockBreakdownData} from 'test/data';
import {parseBreakdownData} from 'event-analysis/utils/utils';
import {render} from '@testing-library/react';

jest.unmock('react-dom');

describe('PercentOfCell', () => {
	const orderedBreakdowns = [
		{dataType: DataTypes.String},
		{dataType: DataTypes.String},
		{dataType: DataTypes.String}
	];

	it('render', () => {
		const data = mockBreakdownData();
		const items = parseBreakdownData(data, orderedBreakdowns);
		const {events} = items[0];

		const {container} = render(
			<PercentOfCell
				compareToPrevious={false}
				events={events}
				totalEvents={data.totalEvents}
			/>
		);

		expect(container).toMatchSnapshot();
	});

	it('render comparing previous', () => {
		const data = mockBreakdownData(true);
		const items = parseBreakdownData(data, orderedBreakdowns);
		const {events} = items[0];

		const {container} = render(
			<PercentOfCell
				compareToPrevious
				events={events}
				totalEvents={data.totalEvents}
			/>
		);

		expect(container.querySelectorAll('ul > li').length).toBe(2);
	});

	it('render comparing previous and segment', () => {
		const data = mockBreakdownData(true, true);
		const items = parseBreakdownData(data, orderedBreakdowns);
		const {events} = items[0];

		const {container} = render(
			<PercentOfCell
				compareToPrevious
				events={events}
				totalEvents={data.totalEvents}
			/>
		);

		expect(container.querySelectorAll('ul > li').length).toBe(4);
	});

	it('render comparing previous and event', () => {
		const data = mockBreakdownData(true, false, true);
		const items = parseBreakdownData(data, orderedBreakdowns);
		const {events} = items[0];

		const {container} = render(
			<PercentOfCell
				compareToPrevious
				events={events}
				totalEvents={data.totalEvents}
			/>
		);

		expect(container.querySelectorAll('ul > li').length).toBe(4);
	});

	it('render two list when comparing segment and event', () => {
		const data = mockBreakdownData(false, true, true);
		const items = parseBreakdownData(data, orderedBreakdowns);
		const {events} = items[0];

		const {container} = render(
			<PercentOfCell
				compareToPrevious
				events={events}
				totalEvents={data.totalEvents}
			/>
		);

		expect(container.querySelectorAll('ul').length).toBe(2);
	});
});
