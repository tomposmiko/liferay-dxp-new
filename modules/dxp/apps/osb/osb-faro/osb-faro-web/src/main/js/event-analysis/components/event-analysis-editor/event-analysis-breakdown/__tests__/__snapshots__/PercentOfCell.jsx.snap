// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PercentOfCell render 1`] = `
<div>
  <ul
    class="percentage-column"
  >
    <li>
      0%
    </li>
  </ul>
</div>
`;
